src/components/_global/BalLink/BalLink.vue

```diff
@@ -1,56 +1,72 @@
-<template>
-  <a v-if="!disabled" :class="[classes]" v-bind="attrs_">
-    <slot />
-  </a>
-  <div v-if="disabled" :class="[classes]" v-bind="attrs_">
-    <slot />
-  </div>
-</template>
+<script lang="ts">
+export default {
+  inheritAttrs: false
+};
+</script>
+
+<script lang="ts" setup>
+import { computed, useAttrs } from 'vue';
+
+/**
+ * TYPES
+ */
+type LinkTag = 'a' | 'router-link';
+
+type Props = {
+  tag: LinkTag;
+  external: boolean;
+  disabled: boolean;
+  noStyle: boolean;
```

Comment:
minor - shouldn't most of these props be optional? (I know they got a default, but was just wondering)

Reply:
By having defaults they are essentially optional. The reason I always set boolean prop defaults to false is:
1. So that the prop value is always a boolean. I'm not 100% what the type will be if I just make it optional but don't pass it, maybe null? I just like being sure.
2. I like to force boolean props to be false so that to use them you just need to pass the prop name in the tag, like:
```vue
<BalLink external noStyle>A link</BalLink>
```

Reply:
It's probably a preference thing really